### PR TITLE
Added New AI Behaviour, And Two Types Of Enemies

### PR DESCRIPTION
**Pull request by MadCoderr**
## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description
i have use Delegation Pattern. [https://unity3d.com/learn/tutorials/topics/navigation/action-and-state-base-classes?playlist=17105](url)
State -> Decision -> Action -> Transition

Why this Pattern. It Allow programmer to efficiently and in manageable set up their AI agent.
With this pattern you can add any kind of decision and action and you AI will be able to follow it.
It's a clean approach so that's why i use it.  

Under The PluggableAI folder you will find State Folder in that there are 3 state.
1- Patrol Chaser State
2- Chase Chaser State
3- Remain In State

Each One is responsible for what their name is being describe from.
Each State have Actions and Transition (Transitions are not themselves Decision but they contain Decision).
Transition is responsible to change State from one to other.
Action is name say are responsible for performing specific action.

All of them are being Called From State Class which is responsible
For Managing Transition Between State And Actions.
And State Class is Being Called From StateController which is a MonoBehaviour Class.

So Back To topic 
There's two enemy AI agent added.
well both of them have same action. But i just change their individual properties.
Magic_caster:  Will Patrol if Saw Player will chase and Attack, Player Die (Disable/Destory) will go back to 
Patrol State. It will for time being Shoot Fireball.

Foot_Soldier:  Will Patrol if Saw Player will chase and Attack, Player Die (Disable/Destory) will go back to 
Patrol State. It will for time being Attack Player With Sword (Cube).

Note:
Both Enemies have Patrol Chaser State as current state.
if You want you can also change the current state of Enemy To Chase So it will Chase the player beside 
of patrolling on Game Start. But for that you need to make another action or re write previous one 
to manually provide ChaseTarget Transform.

Please explain the changes you made here.

## Checklist

Put an `x` in the boxes that apply. 
_You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] The code compiles and has no errors.
- [ ] I have extended the README if necessary
- [x] All added code complies with the set guidelines.
- [x] I have added necessary documentation (if appropriate) to all added code.
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have pulled the latest master into this branch to avoid merge conflicts.
- [ ] I have created new issues for any new bugs that were caused by this pull request.

## New bugs in this PR

Please describe new bugs that have occured in this pull request, also link the issues of said bugs.

## Fixed bugs by this PR

If any, please link the issues of fixed bugs. If none, type "None"

### CODE REVIEW

**Comment by MadCoderr** [comment]
you will find more detail in #dev-talk

File: Cows-HacknSlash/Assets/Scripts/PlayerController/CharController.cs [hunk 1]
```diff
@@ -50,9 +50,12 @@ public class CharController : MonoBehaviour {
 	}
 
     private void OnTriggerEnter(Collider other) {
+        if (other.tag == "Fire_Ball_Enemy") {
```
Comment (GSerum) [inline Cows-HacknSlash/Assets/Scripts/PlayerController/CharController.cs]: Please do not use `==` comparison on strings, rather use `other.tag.Equals("Fire_Ball_Enemy")`

File: Cows-HacknSlash/Assets/Scripts/PluggableAI/Script/Enemy/EnemyManager.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+            _enemyAttack = GetComponent<EnemyAttack>();
+            _enemyAttack.SetUpEnemyType(AttackPosition, EnemyTypeChoice);
+
+            _enemyHealth = GetComponent<EnemyHealth>();
+        }
+
+        private void OnTriggerEnter(Collider other)
+        {
+            if (other.tag == "Fire_Ball")
```
Comment (GSerum) [inline Cows-HacknSlash/Assets/Scripts/PluggableAI/Script/Enemy/EnemyManager.cs]: Here too please

Comment (MadCoderr) [reply Cows-HacknSlash/Assets/Scripts/PluggableAI/Script/Enemy/EnemyManager.cs]: sure will do it

**Comment by MadCoderr** [comment]
Fixed the String quality issue.

**Pull request merged**